sympy/solvers/ode/single.py

```diff
@@ -784,5 +784,71 @@ def _get_general_solution(self, *, simplify: bool = True):
         return [gensol]
 
 
+class SecondNonlinearAutonomousConserved(SinglePatternODESolver):
+    r"""
+    Gives solution for the autonomous second order nonlinear
+    differential equation of the form
+
+    .. math :: f''(x) = g(f(x))
+
+    The solution for this differential equation can be computed
+    by multiplying by `f'(x)` and integrating on both sides,
+    converting it into a first order differential equation.
+
+    Examples
+    ========
+
+    >>> from sympy import Function, symbols, dsolve
+    >>> f, g = symbols('f g', cls=Function)
+    >>> x = symbols('x')
+
+    >>> eq = f(x).diff(x, 2) - g(f(x))
+    >>> dsolve(eq, simplify=False)
+    [Eq(Integral(1/sqrt(C1 + 2*Integral(g(u), u)), (u, f(x))), C2 + x),
+    Eq(Integral(1/sqrt(C1 + 2*Integral(g(u), u)), (u, f(x))), C2 - x)]
+
+    >>> from sympy import exp, log
+    >>> eq = f(x).diff(x, 2) - exp(f(x)) + log(f(x))
+    >>> dsolve(eq, simplify=False)
+    [Eq(Integral(1/sqrt(C1 - 2*u*log(u) + 2*u + 2*exp(u)), (u, f(x))), C2 + x),
+    Eq(Integral(1/sqrt(C1 - 2*u*log(u) + 2*u + 2*exp(u)), (u, f(x))), C2 - x)]
+
+    References
+    ==========
+
+    http://eqworld.ipmnet.ru/en/solutions/ode/ode0301.pdf
+    """
+    hint = "2nd_nonlinear_autonomous_conserved"
+    has_integral = True
+    order = [2]
+
+    def _wilds(self, f, x, order):
+        fy = Wild('fy', exclude=[0, f(x).diff(x), f(x).diff(x, 2)])
+        return (fy,)
+
+    def _equation(self, fx, x, order):
+        fy = self.wilds()[0]
+        return fx.diff(x, 2) + fy
+
+    def _verify(self, fx):
+        fy = self.wilds_match()[0]
+        u = Symbol('u')
+        syms = list(map(lambda sym:sym.name, fy.subs(fx, u).free_symbols))
+        if 'x' in syms:
```

Comment:
This should not be comparing the string representation of symbols. Also what if the symbol name is not `x`?

Reply:
So I should do?
``` python
x = self.ode_problem.sym
if x.name in syms:
.....
```

Reply:
Yes, something like that

Reply:
Wait no.

Reply:
You should not be comparing the symbols by name. Symbols can have the same name but be different symbols e.g. if they have different assumptions or if they are Dummy.

You should compare the symbols directly so:
```
if x in syms:
```

Reply:
I've made the changes. I'll rebase once you review again.

Reply:
Also, I just realized that I don't need to type cast it into a `list`. I'll remove that whenever I push the changes next.